### PR TITLE
Change memo field size and relocate `ciphertexts` field of JoinSplit description

### LINKED ISSUE #1187: Change the memo field size.
We've decided in #85 to increase the memo field size. This ticket is for the actual code change, associated tests.

This should probably be done before the #756 work as a separate isolated PR so we can see any notable performance issues.

### LINKED ISSUE #1190: Implement moving ciphertexts to the end of a JoinSplit description
Per https://github.com/zcash/zips/issues/59 and https://github.com/zcash/zcash/issues/991#issuecomment-223732455

### PR DESCRIPTION
**Pull request by ebfull**
Closes #1190 
Closes #1187

### CODE REVIEW

**Comment by daira** [comment]
I'll review this.

File: src/zcash/Zcash.h [hunk 1]
```diff
@@ -10,7 +10,9 @@
 #define ZC_V_SIZE 8
 #define ZC_RHO_SIZE 32
 #define ZC_R_SIZE 32
-#define ZC_MEMO_SIZE 128
+#define ZC_MEMO_SIZE 512
+
+#define ZC_NOTEPLAINTEXT_SIZE ZC_NOTEPLAINTEXT_LEADING + ZC_V_SIZE + ZC_RHO_SIZE + ZC_R_SIZE + ZC_MEMO_SIZE
```
Comment (daira) [inline src/zcash/Zcash.h]: Put parens around this expression.

**Comment by daira** [comment]
ACK modulo comment.

**Comment by daira** [comment]
ACK. Should not change coverage.

**Comment by bitcartel** [comment]
ACK

**Comment by bitcartel** [comment]
@zkbot r+

**Pull request merged**